### PR TITLE
Add a readme for jenkins jobs

### PR DESCRIPTION
**Pull request by krzyzacy**
which need to be updated soon..

Feel free to add more stuff and do some proof read for me.

### CODE REVIEW

**Comment by ixdy** [comment]
@mtaufen since you asked about this earlier today :)

File: jenkins/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# How to work with jenkins jobs
```
Comment (spxtr) [inline jenkins/README.md]: Nit, capitalize Jenkins everywhere.

**Comment by wonderfly** [comment]
cc/ @kubernetes/goog-image

File: jenkins/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+   For soak job, add an entry to [test-infra/jenkins/job-configs/kubernetes-jenkins/bootstrap-ci-soak.yaml](https://github.com/kubernetes/test-infra/blob/master/jenkins/job-configs/kubernetes-jenkins/bootstrap-ci-soak.yaml)
+
+   For maintenance job, add an entry to [test-infra/jenkins/job-configs/kubernetes-jenkins/bootstrap-maintenance-ci.yaml]()
+
+2. Add foo.sh to [test-infra/jobs](https://github.com/kubernetes/test-infra/tree/master/jobs), which defines your job behavior. You can reference it from other jobs.
+
+3. Add your new job to [test-infra/testgrid/config/config.yaml](https://github.com/kubernetes/test-infra/blob/master/testgrid/config/config.yaml)
+
+4. Make sure all presubmit tests pass.
```
Comment (fejta) [inline jenkins/README.md]: May mention that running `./jenkins/bootstrap_test.py` is an quick way to trigger most of the unit tests.

File: jenkins/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+   
+   For pr job, add an entry to [test-infra/jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-pull.yaml](https://github.com/kubernetes/test-infra/blob/master/jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-pull.yaml)
+
+   For build job, add an entry to [test-infra/jenkins/job-configs/kubernetes-jenkins/bootstrap-ci-commit.yaml](https://github.com/kubernetes/test-infra/blob/master/jenkins/job-configs/kubernetes-jenkins/bootstrap-ci-commit.yaml)
+
+   For soak job, add an entry to [test-infra/jenkins/job-configs/kubernetes-jenkins/bootstrap-ci-soak.yaml](https://github.com/kubernetes/test-infra/blob/master/jenkins/job-configs/kubernetes-jenkins/bootstrap-ci-soak.yaml)
+
+   For maintenance job, add an entry to [test-infra/jenkins/job-configs/kubernetes-jenkins/bootstrap-maintenance-ci.yaml]()
+
+2. Add foo.sh to [test-infra/jobs](https://github.com/kubernetes/test-infra/tree/master/jobs), which defines your job behavior. You can reference it from other jobs.
```
Comment (fejta) [inline jenkins/README.md]: Maybe `foo.sh` and see whether that looks good for the other thinks like bootstrap-ci-soak.yaml

File: jenkins/README.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# How to work with Jenkins jobs
+
+All the job scripts are in the [test-infra/jobs/{job-name}.sh](https://github.com/kubernetes/test-infra/tree/master/jobs) files now.
```
Comment (fejta) [inline jenkins/README.md]: May want to note that we plan on migrating these into yaml files in the near future.

Comment (krzyzacy) [inline jenkins/README.md]: done.

**Comment by mtaufen** [comment]
+1 this. If we could document the env vars in the scripts that configure jobs here as well, that would be extremely useful.

**Pull request merged**